sdk-core/src/test/java/care/data4life/sdk/auth/UserServiceTest.kt

```diff
@@ -0,0 +1,251 @@
+        private const val AUTH_TOKEN = "authToken"
+    }
+
+    private lateinit var apiService: ApiService
+    private lateinit var storage: CryptoSecureStore
+    private lateinit var userService: UserService
+    private lateinit var cryptoService: CryptoService
+    private lateinit var authService: AuthorizationService
+
+    @Before
+    fun setUp() {
+        authService = mockk<AuthorizationService>()
+        userService = mockk<UserService>()
+        storage = mockk<CryptoSecureStore>()
+        apiService = mockk<ApiService>()
+        cryptoService = mockk()
+        userService = spyk(UserService(USER_ALIAS, authService, apiService, storage, cryptoService))
+    }
+
+    @Test
+    fun `isLoggedIn_shouldReturnTrue`() {
+        //given
+        every { authService.isAuthorized(USER_ALIAS) } returns true
+        every { storage.getSecret(AUTH_STATE) } answers {
+            "auth_state".toCharArray()
+        }
+        val mockGCKey = mockk<GCKey>()
+        every { mockGCKey.getSymmetricKey() } returns mockk<GCSymmetricKey>()
+        every { cryptoService.fetchTagEncryptionKey() } returns mockGCKey
+        every { cryptoService.fetchCurrentCommonKey() } returns mockGCKey
+        // when
+        val testSubscriber = userService.isLoggedIn(USER_ALIAS)
+            .test()
+
+        // then
+        testSubscriber
+            .assertNoErrors()
+            .assertValue(true)
+    }
+
+    @Test
+    fun `isLoggedIn_shouldReturnFalse`() {
+        //given
+        every { storage.getSecret(AUTH_STATE) } answers {
+            "auth_state".toCharArray()
+        }
+        val mockGCKey = mockk<GCKey>()
+        every { mockGCKey.getSymmetricKey() } returns mockk<GCSymmetricKey>()
+        every { cryptoService.fetchTagEncryptionKey() } returns mockGCKey
+        every { cryptoService.fetchCurrentCommonKey() } returns mockGCKey
+        // when
+        val testSubscriber = userService.isLoggedIn(USER_ALIAS)
+            .test()
+
+        // then
+        testSubscriber
+            .assertNoErrors()
+            .assertValue(false)
+    }
+
+    @Test
+    fun `getSessionToken should Return True`() {
+        //given
+        every { authService.refreshAccessToken(USER_ALIAS) } returns AUTH_TOKEN
+        // when
+        val testSubscriber = userService.getSessionToken(USER_ALIAS)
+            .test()
+
+        // then
+        testSubscriber
+            .assertNoErrors()
+            .assertValue(AUTH_TOKEN)
+    }
+
+    @Ignore
+    @Test
+    fun `getSessionToken should Throws Error`() {
+        //given
+        every { authService.refreshAccessToken(USER_ALIAS) } returns mockk<String>()
+        // when
+        val testSubscriber = userService.getSessionToken(USER_ALIAS)
+            .test()
+            .await()
+
+        // then
+        testSubscriber
+            .assertNoValues()
+    }
+
+    @Test
+    fun `Given VersionList is fetched, when the call fails return true`() {
```

Comment:
Also I'd like to stay with the pattern, we already introduced, but this just a matter of taste.